### PR TITLE
[seq2seq] fix gpu test

### LINKED ISSUE #348: testBadParentValuesOnGPU failed on GPU
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): tf-nightly-gpu-2.0-preview
- TensorFlow Addons installed from (source, PyPi): source
- TensorFlow Addons version: master
- Python version and type (eg. Anaconda Python, Stock Python as in Mac, or homebrew installed Python etc): tensorflow/tensorflow:custom-op-gpu
- Bazel version (if compiling from source): N/A
- GCC/Compiler version (if compiling from source):
- Is GPU used? (yes/no): yes
- GPU model (if used):

**Describe the bug**

testBadParentValuesOnGPU failed on GPU.

```python
ERROR: testBadParentValuesOnGPU (__main__.GatherTreeTest)
testBadParentValuesOnGPU (__main__.GatherTreeTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/absl/third_party/unittest3_backport/case.py", line 37, in testPartExecutor
    yield
  File "/usr/local/lib/python2.7/dist-packages/absl/third_party/unittest3_backport/case.py", line 162, in run
    testMethod()
  File "/root/.cache/bazel/_bazel_root/e051f2f195071208ea7f081f88730f4f/execroot/__main__/bazel-out/k8-opt/bin/tensorflow_addons/seq2seq/beam_search_ops_test.runfiles/__main__/tensorflow_addons/seq2seq/beam_search_ops_test.py", line 93, in testBadParentValuesOnGPU
    end_token=end_token)
  File "<string>", line 69, in gather_tree
  File "<string>", line 126, in gather_tree_eager_fallback
  File "/usr/local/lib/python2.7/dist-packages/tensorflow_core/python/eager/execute.py", line 67, in quick_execute
    six.raise_from(core._status_to_exception(e.code, message), None)
  File "/usr/local/lib/python2.7/dist-packages/six.py", line 737, in raise_from
    raise value
InvalidArgumentError: Saw invalid parent id -1 at (batch, time, beam) == (0, 0, 1) [Op:GatherTree]
```

**Describe the expected behavior**

**Code to reproduce the issue**

**Other info / logs**

https://source.cloud.google.com/results/invocations/dc386dab-6aeb-4fed-bf01-cc4b8e65c43e/targets/tensorflow_addons%2Fubuntu%2Fgpu%2Fpy2%2Fpresubmit/log

### PR DESCRIPTION
**Pull request by WindQAQ**
Closes #348. Failed because we didn't catch the exception.

### CODE REVIEW

File: tensorflow_addons/seq2seq/beam_search_ops_test.py [hunk 1]
```diff
@@ -89,12 +86,14 @@ def testBadParentValuesOnGPU(self):
...
-                end_token=end_token)
-            self.assertAllEqual(expected_result, self.evaluate(beams))
+            msg = r"parent id -1 at \(batch, time, beam\) == \(0, 0, 1\)"
+            with self.assertRaisesOpError(msg):
+                beams = gather_tree(
+                    step_ids=step_ids,
+                    parent_ids=parent_ids,
+                    max_sequence_lengths=max_sequence_lengths,
+                    end_token=end_token)
+                self.assertAllEqual(expected_result, self.evaluate(beams))
```
Comment (qlzh727) [inline tensorflow_addons/seq2seq/beam_search_ops_test.py]: I don't think this change is correct. The error message is only thrown in the CPU implementation, but not GPU implementation. The raise of this error message means the cudnn kernel is not correctly invoked.

Comment (WindQAQ) [reply tensorflow_addons/seq2seq/beam_search_ops_test.py]: ohoh, I got it. Will invoke it later. But wonder if the implementation on GPU and CPU should raise the same error message.

**Comment by WindQAQ** [comment]
Have built GPU kernel for beam_search_ops. Tests will pass after merging #393.

**Comment by WindQAQ** [comment]
@seanpmorgan @facaiy  it seems that sanity check failed because of timeout. Do you know what exactly happened?  Thanks!
https://source.cloud.google.com/results/invocations/2a2f2ef1-43ac-43d6-973b-bafdf6e3d9b6/log

**Comment by facaiy** [comment]
@WindQAQ Tzu-Wei, can you merge the latest master branch? Perhaps the timeout is related to py36 environment.

**Comment by facaiy** [comment]
sanity-check hanging problem will be fixed by https://github.com/tensorflow/addons/pull/399

**Comment by WindQAQ** [comment]
Thanks 👍

**Pull request merged**